client/my-sites/domains/domain-management/list/item.jsx

```diff
@@ -48,7 +49,7 @@ class ListItem extends React.PureComponent {
 
 	content() {
 		return (
-			<div className="domain-management-list-item__link" onClick={ this.handleClick }>
+			<button className="domain-management-list-item__link" onClick={ this.handleClick }>
```

Comment:
Not sure what's the exact root cause (I'm guessing some ill-written custom CSS that our legacy domains code is plagued with...), but this change of `div` to `button` causes alignment/padding issues:

<img width="1055" alt="Screenshot 2019-11-14 at 17 45 31" src="https://user-images.githubusercontent.com/3392497/68882168-91784d00-0706-11ea-833f-44b30e5021f3.png">

Reply:
Thank you for finding this, @klimeryk ! I switched it back to a `div`, and added a lint ignore to the line. It appears to fix the issue, but please double-check.

Reply:
@sgomes We have CSS resets for the `button` element type that remove all button-ish styling from it, but it's not 100% perfect. Namely, it still has `text-align: center` style, while the default for `div` is `text-align: left`.

Also, `button` elements suffer from some browser bugs. E.g., flex layout inside `button` does vertical spacing/alignment slightly differently from `div`.

I think that two or three lines of CSS can fix the `.domain-management-list-item__link` styling and help us keep the a11y improvement.

Reply:
I reinstated the button. All it took was a couple of `text-align: inherit`.